src/build/mdn-comments.ts

```diff
@@ -117,37 +58,37 @@ function insertComment(
   }
 }
 
+function generateComment(summary: string, name: string): string {
+  // Escape special regex characters in the name
+  const escapedName = name.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+
+  return summary
+    .replace(/\n/g, " ") // remove newlines
+    .replace(
+      // Match optional preceding identifier + dot OR just the name itself
+      new RegExp(`(?:\\b\\w+\\.)?${escapedName}(\\(\\))?`),
+      (match) => `**\`${match}\`**`,
+    )
+    .replace(/"/g, "'")
+    .trim();
+}
+
 export async function generateDescriptions(): Promise<{
   interfaces: { interface: Record<string, any> };
 }> {
-  const stats = await fs.stat(basePath);
-  if (!stats.isDirectory()) {
-    throw new Error(
-      "MDN submodule does not exist; try running `git submodule update --init`",
-    );
-  }
-
+  const content = await readFile(new URL(inputFile), "utf8");
+  const mdn = JSON.parse(content);
   const results: Record<string, any> = {};
-  const indexPaths = await Promise.all(
-    subdirectories.map((dir) => walkDirectory(new URL(dir, basePath))),
-  ).then((res) => res.flat());
-
-  await Promise.all(
-    indexPaths.map(async (fileURL) => {
-      // XXX: Response.json currently causes racy collision
-      if (fileURL.pathname.endsWith("web/api/response/json/index.md")) {
-        return;
-      }
-      const content = await fs.readFile(fileURL, "utf-8");
-      const slug = extractSlug(content);
-      const generatedPath = generatePath(content);
-      if (!slug.length || !generatedPath) {
-        return;
-      }
-
-      const summary = extractSummary(content);
-      insertComment(results, slug, summary, generatedPath);
-    }),
-  );
+  // metadata is an array of objects, each with at least: slug, page-type, summary
+  for (const entry of mdn.data) {
+    const mdnUrl = entry.mdn_url.split("/en-US/docs/")[1];
+    const slugArr = extractSlug(mdnUrl);
+    const path = paths[entry.pageType];
+    const comment = generateComment(entry.summary, slugArr[slugArr.length - 1]);
+    if (!slugArr.length || !path) {
+      continue;
+    }
```

Comment:
```suggestion
    if (!slugArr.length || !path) {
      continue;
    }
    const comment = generateComment(entry.summary, slugArr.at(-1)!);
```

slugArr length check should happen first, and let's use the modern `at()`.